tests/test_ncbi.py

```diff
@@ -0,0 +1,40 @@
+import pytest
+import types
+from fastapi.exceptions import HTTPException
+
+from semantic_search.ncbi import _medline_to_docs, _safe_request, _parse_medline, _get_eutil_records, uids_to_docs, Settings
+
+settings = Settings()
+
+def test_invalid_uid_test():
+    with pytest.raises(HTTPException):
+        uid = ["93846392868"]
+        records = [{"id:": [uid]}]
+        _medline_to_docs(records)
+
+def test_safe_request():
+    eutils_params = {
+        "db": "pubmed",
+        "id": "9887103",
+        "retstart": 0,
+        "retmode": "xml",
+        "api_key": settings.ncbi_eutils_api_key,
+    }
+    url = "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi"
+    assert _safe_request(url, "POST", files = eutils_params).status_code == 200
+
+def test_get_eutil_records():
+    eutil = "efetch"
+    id = "9887103"
+    #checking if generator is returned, need to check integrity of the returned value
+    assert isinstance(_get_eutil_records(eutil, id), types.GeneratorType) == True
```

Comment:
```suggestion
    assert isinstance(_get_eutil_records(eutil, id), types.GeneratorType)
```

No need to check if the value is `True`, `assert` is already asking that